index.bs

```diff
@@ -1549,6 +1549,33 @@ See also:
 
 * [[#naming-is-hard]]
 
+<h3 id="overloading">Use overloading wisely</h3>
+
+If the signature or behaviour of a method will be significantly different
+depending on the arguments passed
+you should define a separate method
+rather than overload a single method.
+
+Overloading a method such that one of its arguments can be
+a single thing or an array is useful.
+Similarly, passing a dictionary of options,
+or using optional arguments,
+are good patterns.
+
+However, if subsequent arguments for a method have to change
+because of the first argument passed,
+or if a method has different behaviour depending on the types of inputs,
+this can hinder code readability,
+and make discovering API functionality more difficult for authors,
+and should be avoided.
+
+<div class="example">
+TODO
+https://xhr.spec.whatwg.org/#interface-formdata
+https://html.spec.whatwg.org/multipage/canvas.html#canvasdrawpath
```

Comment:
Looking at these:

```webidl
undefined fill(optional CanvasFillRule fillRule = "nonzero");
undefined fill(Path2D path, optional CanvasFillRule fillRule = "nonzero");

undefined clip(optional CanvasFillRule fillRule = "nonzero");
undefined clip(Path2D path, optional CanvasFillRule fillRule = "nonzero");

boolean isPointInPath(unrestricted double x, unrestricted double y, optional CanvasFillRule fillRule = "nonzero");
boolean isPointInPath(Path2D path, unrestricted double x, unrestricted double y, optional CanvasFillRule fillRule = "nonzero");

boolean isPointInStroke(unrestricted double x, unrestricted double y);
boolean isPointInStroke(Path2D path, unrestricted double x, unrestricted double y);
```
Is the problem here really that these should have been different methods? Or improper argument ordering? Or that the alternate signatures should instead have been Path2D methods?
I'm unconvinced having a bunch of similar but slightly different methods like `isPointInStrokePath` would have been a good design.

Reply:
I actually like this pattern in the Canvas 2D API. there's an implicit current path that gets used unless you pass in a path. And passing in a path as the first argument feels right too, since it's that path that the operation acts on.